doc/spec/namespace.md

```diff
@@ -0,0 +1,159 @@
+# Namespace
+The namespace represents the full set of named repositories which can be
+referenced by Docker. A repository contains a set of content-addressable blobs
+and tags referencing those blobs. The namespace can be used to discover the
+location and trust information of a repository based on DNS, HTTP requests and
+other means. The namespace is global and owned through the domain at the root of
+the namespace path.
+
+A repository should always be referenced by its fully canonical value. If an
+engine presents a shortened name to a user, that name should be fully
+canonicalized based on rules defined by the engine before contacting a remote
+server. When mirroring repositories, the original name for a repository must be
+used. Changing a repository name is equivalent to copying or moving and will not
+be referenced by the original repository.
+
```

Comment:
We likely have a definition of what names are currently support in the API.  Here we should just have [DNS RFC]/[Existing v2 name format]

Leaving out 1 and 2 is very intentional in this specification. It is should be done by users of the specification. I can ensure to highlight that further, only 3 is a valid repository name in this context.